### ISSUE TITLE
Fix encoding issue when using chef-client cookbook

### ISSUE THREAD

**Issue by rbugajewski** [chef-brigade/dhcp-cookbook#37]
When you use the chef-client cookbook to automatically run the client on each node every n-th minutes the dhcp-cookbook will throw an encoding exception.

The chef-client process runs like this:
```
/opt/chef/embedded/bin/ruby /usr/bin/chef-client -d -P /var/run/chef/client.pid -c /etc/chef/client.rb -i 180 -s 30 -L /var/log/chef/client.log
```
This is the output:
```
[2014-07-04T13:38:57+02:00] INFO: Running queued delayed notifications before re-raising exception
[2014-07-04T13:38:57+02:00] ERROR: Running exception handlers
[2014-07-04T13:38:57+02:00] ERROR: Exception handlers complete
[2014-07-04T13:38:57+02:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
[2014-07-04T13:38:57+02:00] ERROR: service[isc-dhcp-server] (dhcp::_service line 3) had an error: ArgumentError: invalid byte sequence in US-ASCII
[2014-07-04T13:38:57+02:00] ERROR: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

And this is the stack trace:
```
Generated at 2014-07-04 13:38:57 +0200
ArgumentError: service[isc-dhcp-server] (dhcp::_service line 3) had an error: ArgumentError: invalid byte sequence in US-ASCII
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/provider/service/upstart.rb:126:in `==='
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/provider/service/upstart.rb:126:in `block in load_current_resource'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/provider/service/upstart.rb:123:in `open'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/provider/service/upstart.rb:123:in `load_current_resource'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/provider.rb:103:in `run_action'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/resource.rb:637:in `run_action'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/runner.rb:49:in `run_action'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/runner.rb:81:in `block (2 levels) in converge'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/runner.rb:81:in `each'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/runner.rb:81:in `block in converge'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/resource_collection.rb:98:in `block in execute_each_resource'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/resource_collection.rb:96:in `execute_each_resource'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/runner.rb:80:in `converge'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/client.rb:345:in `converge'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/client.rb:431:in `do_run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/client.rb:213:in `block in run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/client.rb:207:in `fork'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/client.rb:207:in `run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/application.rb:217:in `run_chef_client'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/application/client.rb:328:in `block in run_application'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/application/client.rb:317:in `loop'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/application/client.rb:317:in `run_application'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/lib/chef/application.rb:67:in `run'
/opt/chef/embedded/lib/ruby/gems/1.9.1/gems/chef-11.12.8/bin/chef-client:26:in `<top (required)>'
/usr/bin/chef-client:23:in `load'
/usr/bin/chef-client:23:in `<main>'
```

The locales on the Ubuntu node are:
```
# locale
LANG=en_US.UTF-8
LANGUAGE=en_US:en
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
```

[The CHEF-3304 ticket](https://tickets.opscode.com/browse/CHEF-3304) seems to be related. A suggestion is to explicitly set the encoding in each recipe:
```
# encoding: UTF-8
```

When running `chef-client` manually from the CLI everything works fine.

**Comment by tarak** [comment]
+1. Ran into this too.

**Comment by tarak** [comment]
I found out that the upstart config files on ubuntu cause trouble because they contain UTF-8 characters, especially this line:
```
author "Stéphane Graber <stgraber@ubuntu.com>"
```
If this line is changed in both files
```
/etc/init/isc-dhcp-server.conf
```
and
```
/etc/init/isc-dhcp-server6.conf
```
the service is (re-)started without problems.

**Comment by spheromak** [comment]
it's the utf-8 encoding fun. We should run a cleanup on that, or add the ruby encoding headers for ruby 1.9. Tho no one should be running 1.9 at this point ?

**Comment by jmccann** [comment]
I'm not sure it's worth the time to update everything since Chef 12 with ruby 2.1 was released back in late 2014 and it seems ruby 2.0+ defaults to utf-8 encoding.
